examples/form_data/form_data.rs

```diff
@@ -14,15 +13,16 @@ fn main() {
     });
 
     server.post("/confirmation", middleware!{ |req, res|
-        let mut form_data = String::new();
-        req.origin.read_to_string(&mut form_data).unwrap();
+        let form_data = try_with!(res, req.post());
 
-        println!("{}", form_data);
+        println!("{:?}", form_data);
 
         let mut data = HashMap::new();
         data.insert("title", "Confirmation");
-        data.insert("formData", &form_data);
-
+        data.insert("firstname", form_data.get("firstname").unwrap_or("What's your first name?"));
```

Comment:
I guess there's no easy way to get this working with serialization so something like: 
```rust
let foo: SomeType = try_with!(req, form_data.parse())
``` 
would work? Probably something that can be left for future though.